mobilizon_reshare/publishers/platforms/mastodon.py

```diff
@@ -0,0 +1,129 @@
+import pkg_resources
+import requests
+from requests import Response
+
+from mobilizon_reshare.event.event import MobilizonEvent
+from mobilizon_reshare.publishers.abstract import (
+    AbstractPlatform,
+    AbstractEventFormatter,
+)
+from mobilizon_reshare.publishers.exceptions import (
+    InvalidBot,
+    InvalidCredentials,
+    InvalidEvent,
+    InvalidResponse,
+    PublisherError,
+    ServerError,
+    ClientError,
+)
+from mobilizon_reshare.publishers.platforms.utils import uri_join, fqdn_to_uri
+
+
+class MastodonFormatter(AbstractEventFormatter):
+
+    _conf = ("publisher", "mastodon")
+    default_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon.tmpl.j2"
+    )
+
+    default_recap_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon_recap.tmpl.j2"
+    )
+
+    default_recap_header_template_path = pkg_resources.resource_filename(
+        "mobilizon_reshare.publishers.templates", "mastodon_recap_header.tmpl.j2"
+    )
+
+    def validate_event(self, event: MobilizonEvent) -> None:
+        text = event.description
+        if not (text and text.strip()):
+            self._log_error("No text was found", raise_error=InvalidEvent)
+
+    def validate_message(self, message) -> None:
+        if len(message.encode("utf-8")) >= 500:
```

Comment:
why binary? I thought mastodon limit was 840 charachters

Reply:
It's configurable by the admin but the [default](https://docs.joinmastodon.org/user/posting/) is 500. i think we could query for it but not in the formatter (because of credential validations etc happening in the publisher).

Reply:
Yes, definitely.